src/components/views/rooms/UserIdentityWarning.tsx

```diff
@@ -0,0 +1,302 @@
+            }
+        },
+        [membersNeedingApprovalRef, currentPrompt, selectCurrentPrompt],
+    );
+
+    // Initialise the component.  Get the room members, check which ones need
+    // their identity approved, and pick one to display.
+    const loadMembers = useCallback(async (): Promise<void> => {
+        if (!crypto || initialisedRef.current) {
+            return;
+        }
+        // If encryption is not enabled in the room, we don't need to do
+        // anything.  If encryption gets enabled later, we will retry, via
+        // onRoomStateEvent.
+        if (!(await crypto.isEncryptionEnabledInRoom(room.roomId))) {
+            return;
+        }
+        initialisedRef.current = true;
+
+        const gotVerificationStatusUpdate = gotVerificationStatusUpdateRef.current;
+        const membersNeedingApproval = membersNeedingApprovalRef.current;
+
+        const members = await room.getEncryptionTargetMembers();
+
+        for (const member of members) {
+            const userId = member.userId;
+            if (gotVerificationStatusUpdate.has(userId)) {
+                // We're already checking their verification status, so we don't
+                // need to do anything here.
+                continue;
+            }
+            gotVerificationStatusUpdate.set(userId, false);
+            if (await userNeedsApproval(crypto, userId)) {
+                if (!membersNeedingApproval.has(userId) && gotVerificationStatusUpdate.get(userId) === false) {
+                    membersNeedingApproval.set(userId, member);
+                }
+            }
+            gotVerificationStatusUpdate.delete(userId);
+        }
+
+        selectCurrentPrompt();
+    }, [crypto, room, initialisedRef, gotVerificationStatusUpdateRef, membersNeedingApprovalRef, selectCurrentPrompt]);
+
+    loadMembers().catch((e) => {
+        logger.error("Error initialising UserIdentityWarning:", e);
+    });
+
+    // When a user's verification status changes, we check if they need to be
+    // added/removed from the set of members needing approval.
+    const onUserVerificationStatusChanged = useCallback(
+        (userId: string, verificationStatus: UserVerificationStatus): void => {
+            const gotVerificationStatusUpdate = gotVerificationStatusUpdateRef.current;
+
+            if (!initialisedRef.current) {
+                return;
+            }
+
+            if (gotVerificationStatusUpdate.has(userId)) {
+                gotVerificationStatusUpdate.set(userId, true);
+            }
+
+            if (verificationStatus.needsUserApproval) {
+                addMemberNeedingApproval(userId);
+            } else {
+                removeMemberNeedingApproval(userId);
+            }
+        },
+        [initialisedRef, gotVerificationStatusUpdateRef, addMemberNeedingApproval, removeMemberNeedingApproval],
+    );
+    useTypedEventEmitter(cli, CryptoEvent.UserTrustStatusChanged, onUserVerificationStatusChanged);
```

Comment:
do we not need to check that the affected user is a member of the room here?

Reply:
`addMemberNeedingApproval` implicitly does that by calling `room.getMember()`, which will return `null` if the user isn't a member of the room.  (Yes, I'll add a comment saying that.)